### ISSUE TITLE
xpath macro cant accept variable as parametr

### ISSUE THREAD

**Issue by sztosz** [mischov/meeseeks#40]
```
iex(3)> import Meeseeks.XPath
Meeseeks.XPath
iex(4)> path = "//li[last()]"
"//li[last()]"
iex(5)> xpath(path)          
** (FunctionClauseError) no function clause matching in Meeseeks.XPath.xpath/1    
    (meeseeks) expanding macro: Meeseeks.XPath.xpath/1
    iex:5: (file)
iex(5)> xpath("//li[last()]")
%Meeseeks.Selector.Element{
  combinator: %Meeseeks.Selector.Combinator.Children{
    selector: %Meeseeks.Selector.Element{
      combinator: nil,
      filters: [
        %Meeseeks.Selector.XPath.Predicate{
          e: %Meeseeks.Selector.XPath.Expr.Predicate{
            e: %Meeseeks.Selector.XPath.Expr.Function{args: [], f: :last}
          }
        }
      ],
      selectors: [%Meeseeks.Selector.Element.Tag{value: "li"}]
    }
  },
  filters: nil,
  selectors: []
}
iex(6)> xpath(path <> "")    
%Meeseeks.Selector.Element{
  combinator: %Meeseeks.Selector.Combinator.Children{
    selector: %Meeseeks.Selector.Element{
      combinator: nil,
      filters: [
        %Meeseeks.Selector.XPath.Predicate{
          e: %Meeseeks.Selector.XPath.Expr.Predicate{
            e: %Meeseeks.Selector.XPath.Expr.Function{args: [], f: :last}
          }
        }
      ],
      selectors: [%Meeseeks.Selector.Element.Tag{value: "li"}]
    }
  },
  filters: nil,
  selectors: []
}
iex(7)> 
```

It boil's down basically to this, you can't call `xpath(path)` where path is variable containing string, but you can call `xpath(path <> "")`. And I really don't know how to solve this.

**Comment by mischov** [comment]
Good catch, and thank you for your report.

I'll see what I can get figured out, but in the meantime, maybe `xpath("#{path}")` looks a little cleaner than `xpath(path <> "")`

**Comment by mischov** [comment]
@sztosz I'm fixing this, but it's worth noting that using a var instead of a string literal when there's no dynamic values in the string unnecessarily moves the work of compiling the selector from compile time to run time.

Prefer using `xpath("//li[last()]")` when you can.

**Comment by sztosz** [comment]
@mischov Actually I don't mind (at all) how does workaround looks. But `xpath(path <> "")` looks better to me, because at tells you something's not right at first glance ;)

In my case I compute dynamically which node we are we looking for, based on what the "admin" enter's into form. Then, dynamically we calculate path to the given node, and then we check this dynamically computed xpath against few documents. It's a bit complicated but unfortunately has to be done at runtime. The cost of computing which `%Meeseeks.Selector.Element{}` does the xpath points to, for us is actually the least of concerns :joy: 

And thank you for fixing it, it's really easy to workaround, but the workaround wasn't so obvious to me :)

**Comment by mischov** [comment]
@sztosz Fixed in v0.9.2, which is up on Hex now.
